File: Task3/Python/inoutdiamond.py
def generate_diamond(n):
    lines = []
    for i in range(1, n + 1):
        lines.append(' ' * (n - i) + '*' * (2 * i - 1))
    for i in range(n - 1, 0, -1):
        lines.append(' ' * (n - i) + '*' * (2 * i - 1))
    return '\n'.join(lines)

with open('input.txt', 'r') as infile:
    n = int(infile.read().strip())

diamond = generate_diamond(n)

with open('output.txt', 'w') as outfile:
    outfile.write(diamond)

File: Task3/Ruby/inoutdiamond.rb
n = File.read('input.txt').to_i

diamond = ""

(1..n).each do |i|
  diamond << ' ' * (n - i) + '*' * (2 * i - 1) + "\n"
end

(n - 1).downto(1) do |i|
  diamond << ' ' * (n - i) + '*' * (2 * i - 1) + "\n"
end

File.write('output.txt', diamond)

